src/lib/storage/devicegraph_query.rb

```diff
@@ -0,0 +1,64 @@
+require "storage"
+require "storage/disk_size"
+
+module Yast
+  module Storage
+    # THIS (together with the specs) IS THE MAIN CLASS TO DISCUSS IN THE PULL
+    # REQUEST
+    #
+    # We had quite some logic that was initially in SpaceMaker and then moved to
+    # a ProposalDevicegraph refinement for checking things like
+    #  - available space in the candidate disks (discarding small chunks)
+    #  - partitions in the candidate disks (either directly or with part. table)
+    #  - ...
+    #
+    # While writing the tests, I found myself needing to do similar checks, but
+    # not necessarily restricted to the candidate disks
+    #
+    # This class (together with the Disk refinement it relies on) should allow
+    # to do those "high level" queries, setting a filter (only disk_names for
+    # the time being) in advance to set the scope
+    #
+    # The class is now used in the RSpec tests and in the proposal code
+    class DevicegraphQuery
+      using Refinements::Disk
+
+      # Free disk space below this size will be disregarded
+      TINY_FREE_CHUNK = DiskSize.MiB(30)
+
+      attr_reader :devicegraph
+
+      def initialize(devicegraph, disk_names: nil)
```

Comment:
Yes, "Querier" is indeed an ugly word. Probably it doesn't even exist. I strongly suggest to stick with "Query" which is also a noun, it's well-known, widely used, and it wouldn't stick out like a sore thumb.